src/wx/faudio.cpp

```diff
@@ -399,9 +393,8 @@ bool FAudio_Output::init(long sampleRate)
     if (gopts.upmix) {
         // set up stereo upmixing
         FAudioDeviceDetails dd;
-        ZeroMemory(&dd, sizeof(dd));
-        hr = FAudio_GetDeviceDetails(faud, 0, &dd);
-        assert(hr == S_OK);
+        memset(&dd, NULL, sizeof(dd));
+        assert(FAudio_GetDeviceDetails(faud, 0, &dd) == 0);
         float* matrix = NULL;
```

Comment:
Same as above, you can declare `matrix` this way: `std::vector<float> matrix(2 * dd.OutputFormat.Format.nChannels);`
This way you don't have to deal with memory manually.

Reply:
As above, I'll follow up separately.